Simplenote/src/main/java/com/automattic/simplenote/utils/ReviewAccountVerifier.java

```diff
@@ -0,0 +1,74 @@
+package com.automattic.simplenote.utils;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import com.automattic.simplenote.Simplenote;
+import com.automattic.simplenote.models.Account;
+import com.simperium.client.Bucket;
+import com.simperium.client.BucketObjectMissingException;
+
+import static com.automattic.simplenote.models.Account.KEY_EMAIL_VERIFICATION;
+
+/**
+ * When a user login into their account, Simplenote starts indexing the different buckets including Bucket<Account>.
+ * We setup a OnNetworkChangeListener on the account bucket to listen when the bucket is indexed. One of the parameters
+ * of the callback onNetworkChange called is Bucket.ChangeType. If this parameter is INDEX, then we know that the
+ * account bucket was indexed (synchronized). When the account bucket is synchronized, we can check whether the account
+ * was verified or not
+ */
+public class ReviewAccountVerifier implements Bucket.OnNetworkChangeListener<Account> {
+    private enum Status {
+        UNKNOWN,
+        UNVERIFIED,
+        VERIFIED,
+    }
+
+    private final Simplenote simplenote;
+    private Status currentStatus = Status.UNKNOWN;
+
+    public ReviewAccountVerifier(Simplenote simplenote) {
+        this.simplenote = simplenote;
+    }
+
+    public void clearStatus() {
+        currentStatus = Status.UNKNOWN;
+    }
+
+    @Override
+    public void onNetworkChange(Bucket<Account> bucket, Bucket.ChangeType type, String key) {
+        // If we already know the answer, we avoid computing it again
+        if (currentStatus != Status.UNKNOWN) {
+            return;
+        }
+
+        String email = simplenote.getUserEmail();
+        if (type != Bucket.ChangeType.INDEX || email == null) {
+            return;
+        }
+
+        // When a network change of type INDEX is received, it means that the channel finished indexing the bucket, thus
+        // we can check whether the account is verified or not
```

Comment:
side note: it's possible at other times, even _most_ times to check on the verification status. what's important for us here is that we don't check the status before _the first index_ operation or before we positively synchronize with the server. I believe that the Android library _always_ indexes at boot but this is strictly not required by the Simperium protocol.

the primary thing we want to avoid is nagging people when they've already verified because that's annoying and confusing, which is why we wait until we can confidently say "we have all recent changes to the account bucket and the value we read is as good as we can expect."

Reply:
Agree. Let me just be sure the events we are going to handle and make the complete change.